Added MouseEvents AND changed window resize

EtienneBruines opened the pull request:
Also added a demo for the mouse events. See #59.

The function `engi.SetScaleOnResize` allows user to say:
"I want things to become bigger on resize" (i.e. Pong), or
"I want to be able to see more things on resize" (i.e. KeyboardScroller).
See #65.

The `MouseSystem` and `CameraSystem` are changed accordingly.

Also created a separate `audio.go` file for Windows, to
be sure it also builds on Windows. Fixes #121.

paked commented:
Yes. Demo would be great.

On Fri, Nov 27, 2015 at 8:58 AM Etienne Bruines <notifications@github.com>
wrote:

> One simply adds a MouseComponent, and the MouseSystem.
>
> Also see #59 <https://github.com/paked/engi/issues/59>.
>
> Do we also want a demo for this?
> ------------------------------
> You can view, comment on, or merge this pull request online at:
>
>   https://github.com/paked/engi/pull/106
> Commit Summary
>
>    - Added basic Mouse events
>
> File Changes
>
>    - *M* batch.go <https://github.com/paked/engi/pull/106/files#diff-0>
>    (2)
>    - *A* mouse.go <https://github.com/paked/engi/pull/106/files#diff-1>
>    (95)
>    - *M* world.go <https://github.com/paked/engi/pull/106/files#diff-2>
>    (1)
>
> Patch Links:
>
>    - https://github.com/paked/engi/pull/106.patch
>    - https://github.com/paked/engi/pull/106.diff
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/paked/engi/pull/106>.
>

matiwinnetou commented:
:+1:

paked commented:
@EtienneBruines I'm happy to merge this in once you have time to clean up the merge conflicts.

I can make an example at a later date, just add an issue for it.

EtienneBruines commented:
Updated to `master`, and changed some stuff. 

Please review again, and merge when satisfied. The (updated) first-post of this PR might help.  @paked @matiwinnetou

paked commented on audio_windows.go:
```diff
@@ -0,0 +1,41 @@
+	defaultHeightModifier float32 = 1
+)
+
+// AudioComponent is a Component which is used by the AudioSystem
```
This should be defined in a separate `audio.go`, also... Probably should make this file be ommitted from a non-windows build.

EtienneBruines replied:
The `_windows` part of the filename already makes sure of that.

paked replied:
Oh... Derp!

paked commented on mouse.go:
```diff
@@ -0,0 +1,118 @@
+	Leave bool
+}
+
+func (*MouseComponent) Type() string {
```
This needs a comment

EtienneBruines replied:
The `Type()` function? According to `golint`: yes, but tbh, if we were to actually comment *every* `Type()` function, it wouldn't make any sense..?

paked commented on mouse.go:
```diff
@@ -0,0 +1,118 @@
+	mouseDown bool
+}
+
+func (*MouseSystem) Type() string {
```
These functions need comments

paked commented on mouse.go:
```diff
@@ -0,0 +1,118 @@
+	mx := m.mouseX
+	my := m.mouseY
+
+	log.Println(mx, my, space.Position.X, space.Position.Y)
```
Do we need this logging statement?

EtienneBruines replied:
We most certainly do not; good catch, has to be removed.

EtienneBruines commented:
Updated again.

paked commented:
LGTM.
